mono/sgen/sgen-gc.c

```diff
@@ -1901,6 +1906,8 @@ collect_nursery (const char *reason, gboolean is_overflow)
 
 	enqueue_scan_from_roots_jobs (&gc_thread_gray_queue, sgen_nursery_section->data, sgen_nursery_section->end_data, is_parallel ? NULL : object_ops_nopar, is_parallel);
 
+	sgen_workers_flush_deferred_jobs (GENERATION_NURSERY, is_parallel);
```

Comment:
Is there a reason for flushing twice, since we already do it in enqueue_scan_from_roots_jobs ?

Reply:
It flushed but won't signal and then we just signal, but agree, due to the way the calls ended up, we could just flush and signal directly in enqueue_scan_from_roots_jobs. The intent was that enqueue_scan_from_roots_jobs might not be the last function called before dispatching workers, so it flushed since it generates a lot of jobs, but that might be a little over ambitious since we can rely on the auto flush (since I changed it to not signal, just flush). I will correct this, only do explicit flush+signal just before we dispatch workers.

Reply:
Fixed.